src/mongo/mongo.ts

```diff
@@ -125,6 +141,11 @@ export class Model implements IMongoResource {
 			return n !== 0 ? n : a.name.localeCompare(b.name);
 		});
 	}
+
+	private async persistServers() {
+		const secret = JSON.stringify(this._servers.map(s => s.connectionString));
+		await this.azureAccount.credentials.writeSecret(this._credentialsService, this._credentialsAccount, secret);
```

Comment:
I know that Keytar uses [CredWrite](https://msdn.microsoft.com/en-us/library/windows/desktop/aa375187.aspx) function on Windows to store the secret, the function limits the size of the secret to 512 bytes. What if you have many connection strings so the generated JSON size exceeds the limit?

Reply:
Refactored so that we only store one connection string per secret. I tested this on Windows and it was working fine. If you're worried about longer connection strings (not sure if that's possible) I'd prefer to deal with that in a separate issue. I think this will solve it for most customers.